handlers/handlers.go

```diff
@@ -17,50 +17,38 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-type StreamInfo struct {
-	callbackUrl string
-}
-
 type CatalystAPIHandlersCollection struct {
 	MistClient  MistAPIClient
-	StreamCache map[string]StreamInfo
+	StreamCache *StreamCache
 }
 
 func (d *CatalystAPIHandlersCollection) Ok() httprouter.Handle {
-	return func(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	return func(w http.ResponseWriter, _ *http.Request, _ httprouter.Params) {
 		io.WriteString(w, "OK")
 	}
 }
 
-func (d *CatalystAPIHandlersCollection) TranscodeSegment() httprouter.Handle {
-	schema := inputSchemasCompiled["TranscodeSegment"]
-
+// TranscodeSegment takes mpegts segment as input, ingests in Mist under SOURCE_PREFIX streamName, transcodes into RENDITION_PREFIX stream then pushes renditions to specified destination.
```

Comment:
I think this comment is very important, because it explains the whole logic of the `Mist <> Catalyst API` interactions for the transcoding part. Therefore, I suggest to improve it slightly to look something like this:

```
// TranscodeSegment takes mpegts segment as input and asynchronously transcode it into the specified destination.
//
// The process looks as follows:
// 1. Add stream to Mist
// 2. Execute the `MistProceLivepeer` process to perform the following operations:
//   - Start SOURCE_PREFIX stream transcoding in Mist
//     - ...
```